driver/sql/projection_notification_processor_test.go

```diff
@@ -0,0 +1,109 @@
+package sql
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type notificationQueueMock struct {
+	mock.Mock
+	done  chan struct{}
+	queue chan *ProjectionNotification
+}
+
+func (m *notificationQueueMock) Start(done chan struct{}, queue chan *ProjectionNotification) {
+	m.Called(done, queue)
+
+	m.done = done
+	m.queue = queue
+}
+
+func (m *notificationQueueMock) Queue(ctx context.Context, notification *ProjectionNotification) error {
+	args := m.Called(ctx, notification)
+
+	m.queue <- notification
+
+	return args.Error(0)
+}
+
+func (m *notificationQueueMock) ReQueue(ctx context.Context, notification *ProjectionNotification) error {
+	args := m.Called(ctx, notification)
+
+	return args.Error(0)
+}
+
+func TestStartProcessor(t *testing.T) {
+	testCases := []struct {
+		title        string
+		notification func() *ProjectionNotification
+		queueMethod  string
+	}{
+		{
+			"Handle nil notification",
+			func() *ProjectionNotification {
+				return nil
+			},
+			"Queue",
+		},
+		{
+			"Handle new notification",
+			func() *ProjectionNotification {
+				return &ProjectionNotification{
+					No:          1,
+					AggregateID: "abc",
+				}
+			},
+			"ReQueue",
+		},
+		{
+			"Handle retried notification",
+			func() *ProjectionNotification {
+				now := time.Now().Add(time.Millisecond * 200)
+				return &ProjectionNotification{
+					No:          1,
+					AggregateID: "abc",
+					ValidAfter:  now,
+				}
+			},
+			"ReQueue",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.title, func(t *testing.T) {
+			buffer := 1
```

Comment:
```suggestion
			bufferSize := 1
```

Perhaps we can be more clear with the naming here:) wdyt?